### PR TITLE
Make up the test of other bits

### PR DESCRIPTION
**Pull request by joii2020**
Make up the test of 8 16 32 128 512 1024 bits.
Optimization parameters, the test range can be controlled by parameters.
Upload tools used in development.

### CODE REVIEW

File: cases/src/main.rs [hunk 1]
```diff
@@ -73,14 +74,19 @@ fn program_entry(argc: u64, argv: *const *const u8) -> i8 {
             set_verbose(true);
         } else if data.find("--simple").is_some() {
             set_simple(true);
```
Comment (XuJiandong) [inline cases/src/main.rs]: Use "--full" to replace "--simple". By default, it's a simple cases. When "--full" is specified, it becomes full cases.

File: cases/src/main.rs [hunk 2]
```diff
@@ -73,14 +74,19 @@ fn program_entry(argc: u64, argv: *const *const u8) -> i8 {
...
+        } else if data.find("--seed=").is_some() {
```
Comment (XuJiandong) [inline cases/src/main.rs:77]: Don't forget to update CI to enable seed.

File: cases/src/rng_data.rs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+pub static G_DATA: [u8; 512] = [
```
Comment (XuJiandong) [inline cases/src/rng_data.rs]: why is these data used?

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: cases/src/rng.rs [hunk 4]
```diff
@@ -1,201 +1,93 @@
...
-                0x8c91e2fe14041a34,
-                0xc9d368e6546c1f00,
-                0xdfd83d690e5f073e,
-                0x34f2a050c605b6b0,
-                0xf3fbe985738811dd,
-                0x2d21e3da342cd6be,
-                0x31523358d080e093,
-            ],
-        }
+pub fn set_seed(seed: usize) {
```
Comment (XuJiandong) [inline cases/src/rng.rs]: Don't remove BestNumberRng. When seed is not provided, use these numbers.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: cases/src/rng.rs [hunk 5]
```diff
@@ -178,24 +197,94 @@ impl RngCore for BestNumberRng {
...
         Ok(self.fill_bytes(dest))
     }
 }
 
-pub fn new_random_vec(size: usize) -> Vec<u8> {
-    let mut vec = vec![0u8; size];
-    let mut rng = BestNumberRng::default();
-    rng.fill(&mut vec[..]);
-    vec
+pub fn fill_rand_bytes(dest: &mut [u8]) {
```
Comment (XuJiandong) [inline cases/src/rng.rs]: Such code is not necessary if rng trait is implemented.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: cases/src/rng_data.rs (hunk 3 above)
Comment (XuJiandong) [reply cases/src/rng_data.rs]: With PRNG, I think it's not needed.

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: cases/.vscode/launch.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (XuJiandong) [inline cases/.vscode/launch.json]: Remove this.

File: cases/src/main.rs [hunk 7]
```diff
@@ -54,28 +54,40 @@ use ckb_std::syscalls::debug;
...
+
+    let args = unsafe { from_raw_parts(argv, argc as usize) };
+    for i in 0..argc as usize {
+        let data = unsafe { CStr::from_ptr(args[i]) }.to_str().unwrap();
+        if data.find("--case").is_some() {
+            let pos = data.find("--case=").unwrap() + 7;
+            test_pattern = Some(&data[pos..data.len()]);
+        } else if data.find("--verbose").is_some() {
+            set_verbose(true);
+        } else if data.find("--fill").is_some() {
```
Comment (XuJiandong) [inline cases/src/main.rs]: typo: full, not fill

**Review by XuJiandong — CHANGES_REQUESTED** [review]

File: cases/src/main.rs (hunk 7 above)
Comment (joii2020) [reply cases/src/main.rs]: Sorry, correct it immediately

**Review by XuJiandong — APPROVED** [review]

**Pull request merged**